akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/Running.scala

```diff
@@ -315,21 +364,25 @@ private[akka] object Running {
 
     def onSnapshotterResponse(response: SnapshotProtocol.Response): Unit = {
       val signal = response match {
-        case SaveSnapshotSuccess(meta) =>
+        case e @ SaveSnapshotSuccess(meta) =>
+          // # 24698 The deletion of old events are automatic, snapshots are triggered by the SaveSnapshotSuccess.
+          setup.log.debug(s"Persistent snapshot [{}] saved successfully", meta)
+          if (setup.retention.deleteEventsOnSnapshot)
+            internalDeleteEvents(e, state) // if successful, DeleteMessagesSuccess then internalDeleteSnapshots
+          else
+            internalDeleteSnapshots(meta.sequenceNr)
+
           Some(SnapshotCompleted(meta))
-        case SaveSnapshotFailure(meta, ex) =>
-          Some(SnapshotFailed(meta, ex))
-        case DeleteSnapshotSuccess(meta) =>
-          Some(DeleteSnapshotCompleted(DeletionTarget.Individual(meta)))
-        case DeleteSnapshotFailure(meta, ex) =>
-          Some(DeleteSnapshotFailed(DeletionTarget.Individual(meta), ex))
-        case DeleteSnapshotsSuccess(criteria) =>
-          Some(DeleteSnapshotCompleted(DeletionTarget.Criteria(criteria)))
-        case DeleteSnapshotsFailure(criteria, failure) =>
-          Some(DeleteSnapshotFailed(DeletionTarget.Criteria(criteria), failure))
-        case _ => None
+
+        case SaveSnapshotFailure(meta, error) =>
+          setup.log.warning("Failed to save snapshot given metadata [{}] due to [{}]", meta, error.getMessage)
+          Some(SnapshotFailed(meta, error))
+
+        case _ =>
+          None
       }
 
+      setup.log.info("Received snapshot event [{}], returning signal [{}].", response, signal)
```

Comment:
debug level?

Reply:
Good catch, that was from an early investigation, thanks. Updated.

Reply:
Updated.